Mathlib/Tactic/Linter/PedanticLinter.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+#  The "pedantic" linter
+
+The "pedantic" linter emits a warning when the syntax of a command differs substantially
+from the pretty-printed version of itself.
+-/
+open Lean Elab Command
+
+namespace Mathlib.Linter
+
+/--
+The "pedantic" linter emits a warning when the syntax of a command differs substantially
+from the pretty-printed version of itself.
+
+The linter makes an effort to start the highlighting at the first difference.
+However, it may not always be successful.
+It also prints both the source code and the "expected code" in a 5-character radius from
+the first difference.
+-/
+register_option linter.pedantic : Bool := {
+  defValue := true
```

Comment:
Didn't you say this should be `false` by default? I assume this is for testing and will be changed before merging.